pkg/util/compress/compress.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compress
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"crypto/sha256"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+var (
+	defaultFS = afero.NewOsFs()
+)
+
+// PackDir will compress a dir to a tar.gz file
+func PackDir(dir string, writer io.Writer) ([]byte, error) {
```

Comment:
this is duplicate with https://github.com/antrea-io/antrea/blob/684dca3e30a333ff0653ea54df4060ae9e13406e/pkg/apiserver/registry/system/supportbundle/rest.go#L306, can you remove that function and make their callers to call this one?